components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger20.java

```diff
@@ -281,51 +281,71 @@ public String createAPIDefinition(API api) throws APIManagementException {
         String endpoints = environment.getApiGatewayEndpoint();
         String[] endpointsSet = endpoints.split(",");
         Set<URITemplate> uriTemplates = api.getUriTemplates();
+        Set scopes = api.getScopes();
```

Comment:
Assign the proper type being maintained in the Set so that you'd not have to forcefully cast the elements to retrieve the values later when they are accessed.